app/services/preserved_object_handler.rb

```diff
@@ -33,56 +38,99 @@ class PreservedObjectHandler
   validates :druid, presence: true, format: { with: /\A[a-z]{2}[0-9]{3}[a-z]{2}[0-9]{4}\z/ }
   validates :incoming_version, presence: true, numericality: { only_integer: true, greater_than: 0 }
   validates :incoming_size, numericality: { only_integer: true, greater_than: 0 }
+  validates :endpoint, presence: true
 
-  attr_reader :druid, :incoming_version, :incoming_size
+  attr_reader :druid, :incoming_version, :incoming_size, :storage_dir, :endpoint
 
-  def initialize(druid, incoming_version, incoming_size)
+  def initialize(druid, incoming_version, incoming_size, storage_dir)
     @druid = druid
     @incoming_version = version_string_to_int(incoming_version)
     @incoming_size = string_to_int(incoming_size)
+    @storage_dir = storage_dir
+    @endpoint = Endpoint.find_by(storage_location: storage_dir)
   end
 
-  def update_or_create
+  def create
     results = []
-
     if invalid?
       results << result_hash(INVALID_ARGUMENTS, errors.full_messages)
     elsif PreservedObject.exists?(druid: druid)
-      Rails.logger.debug "update #{druid} called and object exists"
-
-      db_object = PreservedObject.find_by(druid: druid)
-      results << update_per_version_comparison(db_object)
+      results << result_hash(OBJECT_ALREADY_EXISTS, 'PreservedObject')
     else
       pp_default = PreservationPolicy.default_preservation_policy
-      PreservedObject.create(druid: druid,
-                             current_version: incoming_version,
-                             size: incoming_size,
-                             preservation_policy: pp_default)
+      po = PreservedObject.create!(druid: druid,
+                                   current_version: incoming_version,
+                                   size: incoming_size,
+                                   preservation_policy: pp_default)
+      status = Status.find_by(status_text: "ok") # TODO: Replace the status_text to get a default status_text
```

Comment:
this is fine;  just want to make sure this is ticketed, as default status is now in master.

Or, you can just do `Status.default_status` and you're done already.

Reply:
👍